kedro/framework/context/context.py

```diff
@@ -202,12 +216,8 @@ def params(self) -> dict[str, Any]:
         except MissingConfigException as exc:
             warn(f"Parameters not found in your Kedro project config.\n{exc!s}")
             params = {}
-
-        if self._extra_params:
-            # Merge nested structures
-            params = OmegaConf.merge(params, self._extra_params)
-
-        return OmegaConf.to_container(params) if OmegaConf.is_config(params) else params  # type: ignore[return-value]
+        _update_nested_dict(params, self._extra_params or {})
+        return params  # type: ignore
```

Comment:
ignore mypy - the warning is coming from
`params = self.config_loader["parameters"]` where it could return `Any` from abstract config loader. There's not much we can do to fix this, for OmegaConfigLoader it already return `Dict[str,Any]`